ludwig/modules/metric_modules.py

```diff
@@ -163,7 +163,7 @@ def compute(self) -> Tensor:
 
     def reset(self):
         super().reset()
-        self.avg = _MeanMetric()
+        self.avg.reset()
```

Comment:
@dantreiman any reason you recreated the `MeanMetric` instance here instead of calling `reset()`? The problem appears to be that the new instance is not placed on the correct device. Calling reset here should (I hope) have the same effect while preserving the device placement, but wanted to double-check with you.